main.tf

```diff
@@ -311,9 +321,19 @@ resource "aws_ecs_service" "default" {
   deployment_maximum_percent         = var.deployment_maximum_percent
   deployment_minimum_healthy_percent = var.deployment_minimum_healthy_percent
   health_check_grace_period_seconds  = var.health_check_grace_period_seconds
-  launch_type                        = var.launch_type
+  launch_type                        = length(var.capacity_provider_strategies) > 0 ? null : var.launch_type
```

Comment:
@k911  Looks good, thanks
Why set `launch_type` to `null` if `capacity_provider_strategies` is provided?
Any docs for that?

Reply:
When launch type is specified capacity provider cannot be used as terraform/awscli will return an error in such a case.

Ref: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ecs-service.html

```
LaunchType
...
If a launchType is specified, the capacityProviderStrategy parameter must be omitted.
```

Ref2: https://docs.aws.amazon.com/cli/latest/reference/ecs/create-service.html

```
--capacity-provider-strategy (list)
...
If a capacityProviderStrategy is specified, the launchType parameter must be omitted. If no capacityProviderStrategy or launchType is specified, the defaultCapacityProviderStrategy for the cluster is used.
```

Reply:
thanks